pkg/repository/prisma/workflow_run.go

```diff
@@ -725,354 +753,432 @@ func workflowRunMetricsCount(ctx context.Context, pool *pgxpool.Pool, queries *d
-			},
-		)
+			jobRunLookupDataParams := make([]dbsqlc.CreateJobRunLookupDataParams, 0)
+			for _, jobRunResult := range createJobRunResults {
 
-		if err != nil {
-			return nil, err
-		}
+				workflowRunId := jobRunResult.WorkflowRunId
+				jobRunId := jobRunResult.ID
+
+				workflowRunOpts := workflowRunOptsMap[sqlchelpers.UUIDToStr(workflowRunId)]
+
+				lookupParams := dbsqlc.CreateJobRunLookupDataParams{
+					Tenantid:    pgTenantId,
+					Triggeredby: workflowRunOpts.TriggeredBy,
+					Jobrunid:    jobRunId,
+				}
+
+				if workflowRunOpts.InputData != nil {
+					lookupParams.Input = workflowRunOpts.InputData
+				}
+
+				jobRunLookupDataParams = append(jobRunLookupDataParams, lookupParams)
 
-		// create the child jobs
-		for _, jobRunId := range jobRunIds {
-			lookupParams := dbsqlc.CreateJobRunLookupDataParams{
-				Tenantid:    pgTenantId,
-				Jobrunid:    jobRunId,
-				Triggeredby: opts.TriggeredBy,
 			}
 
-			if opts.InputData != nil {
-				lookupParams.Input = opts.InputData
+			ids := make([]pgtype.UUID, 0)
+
+			triggeredByIds := make([]string, 0)
+			inputs := make([][]byte, 0)
+			jobRunIds := make([]pgtype.UUID, 0)
+			tenantIds = make([]pgtype.UUID, 0)
+
+			for j := range jobRunLookupDataParams {
+
+				ids = append(ids, sqlchelpers.UUIDFromStr(uuid.New().String()))
+				jobRunIds = append(jobRunIds, jobRunLookupDataParams[j].Jobrunid)
+				tenantIds = append(tenantIds, pgTenantId)
+				triggeredByIds = append(triggeredByIds, jobRunLookupDataParams[j].Triggeredby)
+				inputs = append(inputs, jobRunLookupDataParams[j].Input)
+
 			}
 
-			// create the job run lookup data
-			_, err = queries.CreateJobRunLookupData(
+			_, err = queries.CreateJobRunLookupDatas(
 				tx1Ctx,
 				tx,
-				lookupParams,
+				dbsqlc.CreateJobRunLookupDatasParams{
+					Ids:          ids,
+					Tenantids:    tenantIds,
+					Jobrunids:    jobRunIds,
+					Triggeredbys: triggeredByIds,
+					Inputs:       inputs,
+				},
 			)
 
 			if err != nil {
+				l.Error().Err(err).Msg("failed to create job run lookup data")
 				return nil, err
 			}
 
-			// list steps for the job
-			steps, err := queries.ListStepsForJob(
-				tx1Ctx,
-				tx,
-				jobRunId,
+			steps, err := queries.GetStepsForWorkflowVersion(tx1Ctx, tx,
+				workflowVersionIds,
 			)
 
 			if err != nil {
+				l.Error().Err(err).Msg("failed to get steps for workflow version")
 				return nil, err
 			}
 
-			stepRunIds := make([]pgtype.UUID, 0)
+			var stepActionIds []string
+			var newTenantIds []pgtype.UUID
 
 			for _, step := range steps {
-				err = queries.UpsertQueue(
-					tx1Ctx,
-					tx,
-					dbsqlc.UpsertQueueParams{
-						Tenantid: pgTenantId,
-						Name:     step.ActionId,
-					},
-				)
+				stepActionIds = append(stepActionIds, step.ActionId)
+				newTenantIds = append(newTenantIds, pgTenantId)
+			}
 
-				if err != nil {
-					return nil, err
-				}
+			// think about doing this all in one query
+			err = queries.UpsertQueues(
```

Comment:
I actually think this could be where we're seeing deadlocks (not sure though), let's move UpsertQueues to the QueueStepRun method in the step run engine repository, since we'd like to make the name of the queues dynamic anyway.

Reply:
I'm not really sure what is involved in moving this to QueueStepRun

Reply:
Does Queue Step Run get called further down during the execution?  We lose the concurrency if we do that right ? It's possible we could just do it after this txn commits if that makes things easier? It seems pretty decoupled from the other data?

Reply:
This is quite decoupled, so yeah we could do this in a separate tx, but I'd still prefer to move it to QueueStepRun for now so that we can make the queue names dynamic.

Reply:
@abelanger5 I definitely don't understand enough about what is happening here to make the change right now. tbh I'm not 100% sure how the queues work.

Reply:
ok moved it to *stepRunEngineRepository) QueueStepRun